tutorials/basic-topics/sorting/bubble-sort.md

```diff
@@ -1,14 +1,156 @@
 ---
+
 title: 'Bubble Sort'
-description: ''
+
+description: 'It is sorting algorithm based on the idea of repeatedly comparing pairs of adjacent elements and then swapping their positions if they exist in the wrong order.'
+
 hide_table_of_contents: true
-draft: true
+
 keywords:
-  - leetcode
-  - tutorial
-  - sorting
-  - algorithm
-  - bubble sort
+
+- leetcode
+
+- tutorial
+
+- sorting
+
+- algorithm
+
+- bubble sort
+
 ---
 
-<TutorialAuthors names="@TBC"/>
+  
+
+<TutorialAuthors  names="@RadhikaChhabra17"/>
+
+  
+
+## Overview
+
+  
+
+Bubble Sort is a sorting algorithm which compares the adjacent elements and swap their positions if they are placed in wrong order. At max, we need to compare adjacent elements for (n-1) iterations where n is the size of array to be sorted. At the end of each iteration, larger(or smaller, as required) value is sorted and placed at correct positions.
+
+  
+
+The syntax that we are using is `cpp`, please refer to your own language of preference if needed.
+
+Algorithm :
+1. Make two nested loop, the 1st loop would be for number of pass the algorithm woud run, total n-1 passes and second would be for comparison in that pass. In each pass, one element is sorted (largest or smallest, as required) and placed in correct position and rest are compared in further passes.
+2. If one element is bigger(or smaller in decreasing order case) than its next element, then both should be swapped. 
+
+
+  
+
+Consider the example of unsorted list and see how the algorithm works.
+
+  
+
+```c++
+
+arr = {3,6,4,2,5}
+
+Working of bubble sort on this example : 
+
+1st pass =
+	i = 0 -> 3,6,4,2,5
+	i = 1 -> 3,4,6,2,5
+	i = 2 -> 3,4,2,6,5
+	i = 3 -> 3,4,2,5,6
+
+2nd pass = 
+	i = 0 -> 3,4,2,5,6
+	i = 1 -> 3,2,4,5,6
+	i = 2 -> 3,2,4,5,6
+
+3rd pass = 
+	i = 0 -> 2,3,4,5,6
+	i = 1 -> 2,3,4,5,6
+
+4th pass =
+	i = 0 -> 2,3,4,5,6
+
+Code :
+
+void bubblesort(vector<int> &arr) {
+	int n = arr.size();
+	bool check = true;
+	for(int i=0; i<n-1 && check; i++) {
+		check = false;
+		for(int j=0; j<n-i-1; j++) {
+			if(arr[j] > arr[j+1]) {
+				int temp = arr[j];
+				arr[j] = arr[j+1];
+				arr[j+1] = temp;
+				check = true;
+			}
+		}
+	}
+}
+
+```
+### Time and Space Complexity
+
+Time Complexity :
+For first iteration, loop will run (n-1) times, for second it will run (n-2) times and so on.
+So, Worst Case Time Complexity = (n-1) + (n-2) + (n-3) + ... + 1 = O(n^2)
+If there is no change in any iterations, next iterations will not occurs, this reduces the time complexity.
+
+Space Complexity :
+Since there is no extra space, Space Complexity = O(1)
+This shows that it is an inline sorting.
```

Comment:
1. these are packed together. Verify it in your local by running the app. (Not looking at markdown editor)
2. as stated in checklist, you need to apply LaTex on variables / formula / time & space complexity